### PR TITLE
Tested annotate_from_gff works from gff3 files

### PR DESCRIPTION
**Pull request by ChristopherBradley**
Currently ignoring all attributes except Parent and ID which are necessary for constructing the annotation tree. Might want to look into storing the attributes.

### CODE REVIEW

File: tests/test_core/test_alignment.py [hunk 1]
```diff
@@ -888,6 +894,37 @@ def test_annotate_from_gff(self):
...
+
+            # you can annotate a sequence that is part of an alignment
+            seq_name = "P0A7B8"
+            aln = self.Class({seq_name: seq})
+            aln.annotate_from_gff(gff3_path)
+            aln_seq = aln.named_seqs[seq_name]
+            if not hasattr(aln_seq, "annotations"):
+                aln_seq = aln_seq.data
+            self.assertEqual(len(aln_seq.annotations), 35)
+            matches = [m for m in aln_seq.get_annotations_matching("*")]
```
Comment (ChristopherBradley) [inline tests/test_core/test_alignment.py]: matches = ... is redundant. Will remove next push.

File: tests/test_core/test_alignment.py [hunk 2]
```diff
@@ -888,6 +894,50 @@ def test_annotate_from_gff(self):
...
+            matches = [m for m in aln_seq_3.get_annotations_matching("*")]
+            self.assertFalse("-" in matches[0].get_slice())
+
+    def test_annotate_from_gff3(self):
+        """annotate_from_gff should work on data from gff3 files"""
+
+        if self.Class == Alignment:
+            from cogent3.parse.fasta import FastaParser
+
+            fasta_path = "data/gff3_test.fasta"
```
Comment (GavinHuttley) [inline tests/test_core/test_alignment.py]: don't specify path's as full string, use `os.path.join()`

File: tests/test_core/test_alignment.py [hunk 3]
```diff
@@ -888,6 +894,50 @@ def test_annotate_from_gff(self):
...
+            gff3_path_nested = "data/gff3_nested.gff3"
+            aln.annotate_from_gff(gff3_path_nested)
+            aln_seq = aln.named_seqs[seq_name]
+            aln_seq = aln_seq.data
+            matches = [m for m in aln_seq.get_annotations_matching("operon")]
+            self.assertEqual(len(matches), 1)
+            nested_matches = matches[0].get_annotations_matching("mRNA")
+            self.assertEqual(len(nested_matches), 2)
+            nested_matches2 = nested_matches[0].get_annotations_matching("exon")
+            self.assertEqual(len(nested_matches2), 5)
```
Comment (GavinHuttley) [inline tests/test_core/test_alignment.py]: suggest adding `get_annotations_matching("*", extend_query=True)`

**Comment by GavinHuttley** [comment]
subsumed within another PR

**Pull request closed without merging**